File: CarRentalApp.Business/Models/Order.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CarRentalApp.Business.Models
{
    public class Order
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public int CarId { get; set; }
        public Car Car { get; set; }
        public DateTime BeginDate { get; set; }
        public DateTime EndDate { get; set; }
        public string Comment { get; set; }
    }
}

File: CarRentalApp/ClientApp/app/components/orders/order.ts
import { User } from '../users/user';
import { Car } from '../cars/car';
export class Order {
    constructor(
        public id?: number,
        public User?: User,
        public userId?: number,
        public Car?: Car,
        public carId?: number,
        public beginDate?: string,
        public endDate?: string,
        public comment?: string
    ) { }
}